File: Server/src/main/java/core/game/content/dialogue/GerrantDialogue.java
package core.game.content.dialogue;

import core.game.node.entity.npc.NPC;
import core.plugin.Initializable;
import core.game.node.entity.player.Player;

/**
 * Represents the dialogue plugin used for the gerrant npc.
 * @author 'Vexia
 * @version 1.0
 */
@Initializable
public final class GerrantDialogue extends DialoguePlugin {

	/**
	 * Constructs a new {@code GerrantDialogue} {@code Object}.
	 */
	public GerrantDialogue() {
		/**
		 * empty.
		 */
	}

	/**
	 * Constructs a new {@code GerrantDialogue} {@code Object}.
	 * @param player the player.
	 */
	public GerrantDialogue(Player player) {
		super(player);
	}

	@Override
	public DialoguePlugin newInstance(Player player) {
		return new GerrantDialogue(player);
	}

	@Override
	public boolean open(Object... args) {
		npc = (NPC) args[0];
		interpreter.sendDialogues(npc, FacialExpression.HALF_GUILTY, "Welcome! You can buy fishing equipment at my store.", "We'll also buy anything you catch off you.");
		stage = 0;
		return true;
	}

	@Override
	public boolean handle(int interfaceId, int buttonId) {
		switch (stage) {
		case 0:
			interpreter.sendOptions("Choose an option:", "Let's see what you've got then.", "Sorry, I'm not interested.");
			stage = 1;
			break;
		case 1:
			switch (buttonId) {
			case 1:
				interpreter.sendDialogues(player, FacialExpression.HALF_GUILTY, "Let's see what you've got then.");
				stage = 10;
				break;
			case 2:
				interpreter.sendDialogues(player, FacialExpression.HALF_GUILTY, "Sorry, I'm not interested.");
				stage = 20;
				break;

			}
			break;
		case 10:
			end();
			npc.openShop(player);
			break;
		case 20:
			end();
			break;
		}
		return true;
	}

	@Override
	public int[] getIds() {
		return new int[] { 558 };
	}
}

File: Server/src/main/content/region/asgarnia/burthorpe/quest/heroesquest/GerrantDialogue.kt
package content.region.asgarnia.burthorpe.quest.heroesquest

import core.api.*
import core.game.dialogue.DialogueBuilder
import core.game.dialogue.DialogueBuilderFile
import core.game.dialogue.DialoguePlugin
import core.game.dialogue.FacialExpression
import core.game.node.entity.player.Player
import core.plugin.Initializable
import org.rs09.consts.Items
import org.rs09.consts.NPCs

@Initializable
class GerrantDialogue (player: Player? = null) : DialoguePlugin(player) {
    override fun newInstance(player: Player): DialoguePlugin {
        return GerrantDialogue(player)
    }
    override fun handle(interfaceId: Int, buttonId: Int): Boolean {
        openDialogue(player, GerrantDialogueFile(), npc)
        return false
    }
    override fun getIds(): IntArray {
        return intArrayOf(NPCs.GERRANT_558)
    }
}
class GerrantDialogueFile : DialogueBuilderFile() {
    override fun create(b: DialogueBuilder) {

        b.onPredicate { _ -> true }
        .npc(FacialExpression.HAPPY, "Welcome! You can buy fishing equipment at my store.", "We'll also buy anything you catch off you.")
                .options()
                .let { optionBuilder ->
                    optionBuilder.option_playerl("Let's see what you've got then.")
                            .endWith { _, player ->
                                openNpcShop(player, npc!!.id)
                            }

                    optionBuilder.option_playerl("Sorry, I'm not interested.")
                            .end()

                    optionBuilder.optionIf("I want to find out how to catch a lava eel.") { player -> return@optionIf getQuestStage(player, HeroesQuest.questName) >= 1 }
                            .playerl("I want to find out how to catch a lava eel.")
                            .npcl("Lava eels, eh? That's a tricky one, that is. You'll need a lava-proof fishing rod. The method for making this would be to take an ordinary fishing rod, and then cover it with fire-proof blamish oil.")
                            .branch { player ->
                                return@branch if (inInventory(player, Items.BLAMISH_SNAIL_SLIME_1581)) { 1 } else { 0 }
                            }.let { branch ->
                                branch.onValue(1)
                                        .npcl("Of course, you knew that already.")
                                        .playerl("So where can I fish lava eels?")
                                        .npcl("Taverley dungeon or the lava maze in the Wilderness.")
                                        .end()

                                branch.onValue(0)
                                        .npcl("You know... thinking about it... I may have a jar of blamish slime around here somewhere. Now where did I put it?")
                                        .linel("Gerrant searches around a bit.")
                                        .betweenStage { df, player, _, _ ->
                                            addItemOrDrop(player, Items.BLAMISH_SNAIL_SLIME_1581)
                                        }
                                        .npcl("Aha! Here it is! Take this slime, mix it with some Harralander and water and you'll have the blamish oil you need to treat your fishing rod.")
                                        .end()
                            }

                }
    }
}